ui/apiserver/apiserver.go

```diff
@@ -336,7 +336,7 @@ func (api *ApiServer) runIlabChatCommand(question, context string) (string, erro
 		cmd = exec.Command("echo", cmdArgs...)
 		api.logger.Infof("Running in test mode: %s", commandStr)
 	} else {
-		modelName, err := api.fetchModelName(true)
+		modelName, err := api.fetchModelName(true, api.preCheckEndpointURL)
```

Comment:
This change allows for using different models in precheck and precheck scoring via fetch model name, it will go and lookup the model by the endpoint, rathe than pulling it from a single place